api/mapping/urls.py

```diff
@@ -7,16 +7,23 @@
 
 urlpatterns = [
     path('', views.home, name='home'),
-    path('tables/', views.ScanReportTableListView.as_view(), name='tables'),
-    path('fields/', views.ScanReportFieldListView.as_view(), name='fields'),
-    path('fields/<int:pk>/update/', views.ScanReportFieldUpdateView.as_view(), name='scan-report-field-update'),
-    path('fields/<int:pk>/create_mapping/', views.AddMappingRuleFormView.as_view(), name='create-mapping-form'),
-    path('fields/<int:pk>/mapping_rules/', views.StructuralMappingListView.as_view(), name='view-structural-mapping'),
-    path('fields/<int:pk>/mapping_rules/delete', StructuralMappingDeleteView.as_view(), name='structural-mapping-delete'),
-    path('values/', views.ScanReportValueListView.as_view(), name='values'),
-    path('scanreports/', views.ScanReportListView.as_view(), name='scan-report-list'),
-    path('scanreports/<int:pk>/mapping_rules', views.StructuralMappingTableListView.as_view(), name='tables-structural-mapping'),
-    path('scanreports/create/', views.ScanReportFormView.as_view(), name='scan-report-form'),
+    path('tables/', login_required(views.ScanReportTableListView.as_view()), name='tables'),
+    path('fields/', login_required(views.ScanReportFieldListView.as_view()), name='fields'),
+    path('fields/<int:pk>/update/', login_required(views.ScanReportFieldUpdateView.as_view()), name='scan-report-field-update'),
+    path('fields/<int:pk>/create_mapping/', login_required(views.AddMappingRuleFormView.as_view()), name='create-mapping-form'),
+    path('fields/<int:pk>/mapping_rules/', login_required(views.StructuralMappingListView.as_view()), name='view-structural-mapping'),
+    path('fields/<int:pk>/mapping_rules/delete', login_required(StructuralMappingDeleteView.as_view()), name='structural-mapping-delete'),
+    path('values/', login_required(views.ScanReportValueListView.as_view()), name='values'),
+    path('scanreports/', login_required(views.ScanReportListView.as_view()), name='scan-report-list'),
+    path('scanreports/<int:pk>/mapping_rules', login_required(views.StructuralMappingTableListView.as_view()), name='tables-structural-mapping'),
+    path('scanreports/create/', login_required(views.ScanReportFormView.as_view()), name='scan-report-form'),
```

Comment:
there might have been a master merge issue here?

I believe `login_required` got removed by @vpnu and instead we used `@login` decorator in `view.py`

Reply:
Yeah it was removed a while ago right after the documents page was added

Reply:
now we just need @method_decorator(login_required,name='dispatch') before the view class

Reply:
Yeah the decorator makes more sense, I don't recall `login_required` being removed though. Could well be a master merge issue.

Reply:
Yeh @simon-tarr  i think it just needs these removing / rebase/remerge with the master and then i can approve and merge to master when that is done